### PR TITLE
Version of the API that should be simple, extensible and ready for MVP.

### PR DESCRIPTION
**Pull request by mounirlamouri**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -185,6 +185,23 @@
     </section>
     <section>
       <h2>
+        Permission description
```
Comment (domenic) [inline index.html]: Probable typo: s/description/descriptor

File: index.html [hunk 2]
```diff
@@ -345,6 +353,10 @@
         </dt>
       </dl>
       <p>
+        A <code>PermissionStatus</code> instance has an associated
+        <var>permission</var>.
```
Comment (domenic) [inline index.html]: Would be clearer I think if you said "that is a `PermissionDescriptor`"

File: index.html [hunk 3]
```diff
@@ -365,8 +377,8 @@
       <ol>
         <li>Let <var>status</var> be a <code>PermissionStatus</code> instance.
         </li>
-        <li>Set the <code>permission</code> attribute to
-        <code>permission</code>.
+        <li>Set the <code>permission</code> as the <var>status</var> associated
```
Comment (domenic) [inline index.html]: This sentence is confusing, possible typo?

File: index.html [hunk 4]
```diff
@@ -446,12 +453,26 @@
         the following steps:
       </p>
       <ol>
+        <li>If <var>permission.name</var> has an <a>associated
+        PermissionDescriptor</a>, convert the underlying ECMAScript object to
+        the <a>associated PermissionDescriptor</a> dictionary as <a href=
+        'http://heycam.github.io/webidl/#es-dictionary'>described</a> in
+        [[!WEBIDL]], then:
+          <ul>
+            <li>If that operation failed, return a rejected <a>Promise</a> with
```
Comment (domenic) [inline index.html]: "a Promise rejected with a TypeError"

File: index.html [hunk 5]
```diff
@@ -496,7 +517,7 @@
     document.getElementById('chat-notification-button').disabled = (permission.status == 'denied');
   }
 
-  navigator.permissions.query('notifications').then(function(result) {
+  navigator.permissions.query({name:'notifications'}).then(function(result) {
     updateNotificationButton(result);
```
Comment (domenic) [inline index.html]: Given that `result` is basically just `status` + an event handler, maybe just pass `result.status` here and below.

**Comment by domenic** [comment]
LGTM, only nits.

File: index.html (hunk 1 above)
Comment (marcoscaceres) [inline index.html]: descriptor sounds good.

File: index.html [hunk 6]
```diff
@@ -185,6 +185,23 @@
...
       <h2>
+        Permission description
+      </h2>
+      <dl class='idl' title='dictionary PermissionDescriptor'>
+        <dt>
+          required PermissionName <a>name</a>
+        </dt>
+      </dl>
+      <p>
+        A permission is defined by a name and other properties that depends on
```
Comment (marcoscaceres) [inline index.html]: s/depends/depend

File: index.html [hunk 7]
```diff
@@ -185,6 +185,23 @@
...
+        the name. The simplest permissions require only a name but some others
```
Comment (marcoscaceres) [inline index.html]: s/but/, but/

File: index.html [hunk 8]
```diff
@@ -185,6 +185,23 @@
...
+        will require more information. In which case, they should have an
+        <dfn>associated PermissionDescriptor</dfn> dictionary that inherits
```
Comment (marcoscaceres) [inline index.html]: Don't say what it inherits from. It's redundant. Maybe say:

```HTML
In which case, they will have an <dfn>associated permission descriptor</dfn>, which is represented by the <code>PermissionDescriptor</code> dictionary. 
```

File: index.html (hunk 4 above)
Comment (marcoscaceres) [inline index.html]: :+1:

File: index.html (hunk 5 above)
Comment (marcoscaceres) [inline index.html]: agree.

File: index.html [hunk 9]
```diff
@@ -474,7 +495,7 @@
       </p>
       <pre class='example highlight'>
 &lt;script&gt;
-  navigator.permissions.query('geolocation').then(function(result) {
+  navigator.permissions.query({name:'geolocation'}).then(function(result) {
```
Comment (marcoscaceres) [inline index.html]: Nit... we can modernize the examples later tho:
```
navigator.permissions
  .query({name:'geolocation'})
  .then((result) => {
```

File: index.html (hunk 1 above)
Comment (mounirlamouri) [inline index.html]: Done.

File: index.html (hunk 2 above)
Comment (mounirlamouri) [inline index.html]: Done.

File: index.html (hunk 3 above)
Comment (mounirlamouri) [inline index.html]: Rephrased a bit.

File: index.html (hunk 4 above)
Comment (mounirlamouri) [inline index.html]: Done.

File: index.html (hunk 5 above)
Comment (mounirlamouri) [inline index.html]: Done.

File: index.html (hunk 6 above)
Comment (mounirlamouri) [inline index.html]: Done.

File: index.html (hunk 7 above)
Comment (mounirlamouri) [inline index.html]: Done.

File: index.html (hunk 8 above)
Comment (mounirlamouri) [inline index.html]: I want to underline that it needs to be a specialized ```PermissionDescriptor```. I will eat the redundancy for the sake of clarity for now. Let see if that still feels redundant later.

**Comment by mounirlamouri** [comment]
Thank you @marcoscaceres and @domenic for the review! :)

**Comment by marcoscaceres** [comment]
No problem. Will be expecting check in mail.

**Pull request merged**